core/src/main/scala/tofu/Context.scala

```diff
@@ -113,16 +113,30 @@ object Context {
     * object MyContext extends Context.Companion[MyContext]
     * }}}
     */
-  trait Companion[C] {
+  trait Companion[C] extends ContextInstances[C] {
     type Has[F[_]] = WithContext[F, C]
 
-    implicit def tofuPromoteContextStructure[F[_], A](implicit
+    abstract class WithContextOpaque[F[_], C0] extends WithContext[F, C0]
```

Comment:
Why did you make it an inner class? It could be a concrete high-level class, as we usually do in tofu:
```scala
class WithContextContains[F[_], C, A](implicit wc: WithContext[F, C],  field: C Contains A)
  extends WithContext[F, A] {
  private val wa =  wc.extract(field) 
  def functor: Functor[F] = wa.functor
  def context: F[A]       = wa.context
}
```

And then in trait.
```scala
implicit def promoteContextStructure[F[_], A](implicit withContext: WithContext[F, C],
    field: C Contains A
): WithContextContains[F, C, A] =
    new WithContextContains[F, C, A]
```

Reply:
@catostrophe, my bad, we'd better keep style consistent.